### PR TITLE
Remove Pkg from dependencies

### PR DESCRIPTION
**Pull request by sumiya11**
A PR for #1895. 

I don't know if `Base.get_pkgversion_from_path` covers all the cases that were previously handled.

### CODE REVIEW

**Review by lgoettgens — CHANGES_REQUESTED** [review]
This does not work on julia 1.6 (the compat bound of Nemo)

**Comment by sumiya11** [comment]
Would it be reasonable to do `if VERSION < 1.11 then "old version()" else "new version()" end` ?

**Comment by lgoettgens** [comment]
`Base.get_pkgversion_from_path` is non-public, so shouldn't be relied on

File: src/Nemo.jl [hunk 1]
```diff
@@ -383,17 +381,7 @@ end
...
-  version() = "building"
-else
-  ver = deps[Base.UUID("2edaba10-b0f1-5616-af89-8c11ac63239a")]
-  if occursin("/dev/", ver.source)
-    version() = VersionNumber("$(ver.version)-dev")
-  else
-    version() = VersionNumber("$(ver.version)")
-  end
-end
+version() = Base.get_pkgversion_from_path(pkgdir)
```
[the remark this answers is not in the document]
Comment (fingolfin) [reply src/Nemo.jl:384]: In Oscar.jl we instead use this code to determine if it is a dev version (but this is also far from perfect -- and it uses `Pkg`):
```
const is_dev = (function()
        deps = Pkg.dependencies()
        if Base.haskey(deps, PROJECT_UUID)
          if deps[PROJECT_UUID].is_tracking_path
            return true
          end
        end
        return occursin("-dev", lowercase(string(VERSION_NUMBER)))
    end)()
```

**Comment by fingolfin** [comment]
@sumiya11 I have no idea and in any case, none of this is documented AFAIK. I am not concerned with imitating current behavior 1-1. Rather, the general idea, which is to be able to detect reliably which version a user reporting an issue really is using: i.e. either a specific stable release, and which; or some git version (ideally identified by commit). Just like the Julia banner does (perhaps one could look at its current code for inspiration).

**Comment by sumiya11** [comment]
> I am not concerned with imitating current behavior 1-1. Rather, the general idea, which is to be able to detect reliably which version a user reporting an issue really is using: i.e. either a specific stable release, and which; or some git version (ideally identified by commit). Just like the Julia banner does (perhaps one could look at its current code for inspiration).

Thank you for the clarification.

If so, why not ask the user for the output of `Pkg.status()` instead of relying on the error-prone `version` ?

**Comment by thofma** [comment]
It is mainly for the banner which includes the version number. Maybe they fix the Pkg loading regression in 1.11.1. It is not the only thing that got slower.

**Pull request closed without merging**